include/mqtt/broker/security.hpp

```diff
@@ -0,0 +1,492 @@
+// Copyright Wouter van Kleunen 2021
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(MQTT_BROKER_SECURITY_HPP)
+#define MQTT_BROKER_SECURITY_HPP
+
+#include <string>
+
+#include <mqtt/broker/broker_namespace.hpp>
+#include <mqtt/broker/subscription_map.hpp>
+#include <mqtt/optional.hpp>
+
+#include <map>
+#include <set>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/optional.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+
+#if MQTT_USE_TLS
+#include <openssl/evp.h>
+#endif
+
+MQTT_BROKER_NS_BEGIN
+
+struct security
+{
+
+    struct authentication
+    {
+        enum class method {
+            password,
+            client_cert,
+            anonymous
+        };
+
+        method method_;
+        optional<std::string> password;
+
+        authentication(method method_ = method::password, optional<std::string> const &password = optional<std::string>())
+            : method_(method_), password(password)
+        { }
```

Comment:
```suggestion
        authentication(method method_ = method::password, optional<std::string> const& password = optional<std::string>())
            : method_(method_), password(password)
        {}
```

Reply:
And move up to line 45.
Please don't mix member functions and member variables. And place member functions first.